README.md

```diff
@@ -12,23 +12,359 @@
+testing. All peer deployments should be instrumented with prometheus-friendly
+metrics, should they want to have data collected. This is described in greater
+detail in the [scenario runners](#scenario-runners) section. Clusters
+specifying a deployment of the `prometheus` plugin will automatically have
+metrics collected.
+
+## Usage
+
+Testlab is a wrapper over [nomad](https://www.nomadproject.io/)'s golang API,
+making it easy to deploy pre-configured networks of p2p applications.
+
+Most users of testlab need only concern themselves with two concepts, the
+[deployment configuration](#deployment-configuration), and
+[scenarios](#scenario-runners). Users wishing to add testlab support for their
+own daemons will need to understand the [node API](#node-api), as well.
+
+### CLI
+
+The testlab CLI depends on the presence of the standard environment variables to
+connect out to your Nomad and Consul clusters. If any are ommitted, the
+defaults, as defined by Hashicorp, will be applied. The defaults are typically
+usable in development.
+
+- [Nomad environment variables](https://github.com/hashicorp/nomad/blob/v0.9.3/api/api.go#L217)
+- [Consul environment variables](https://github.com/hashicorp/consul/blob/v1.5.2/api/api.go#L24)
+
+Furthermore, users can optionally provide a path in the environment variable
+`TESTLAB_ROOT` to define where the testlab metadata will be stored. This
+defaults to `/tmp/testlab`. **NOTE**: In order to have multiple testlab
+topologies in flight at the same time, one must define different `TESTLAB_ROOT`s
+for each topology.
+
+The testlab CLI has two commands:
+
+- `testlab start <json configuration>`
+  Parses, evaluates for correctness, and attempts to deploy a topology as
+  defined by the provided json configuration file.
+- `testlab stop`
+  Stops the current running topology, identified by its `TESTLAB_ROOT`.
+
+### Deployment Configuration
+
+The entrypoint for most projects using the testlab will be their deployment
+configuration, a JSON document declaring the desired network configuration.
+It's broken into the following top level sections:
+
+#### `Name: string`
+
+The name of the deployment. This will become a prefix to all tasks launched
+in the testlab.
+
+#### `Options: object`
+
+Cluster-wide options to apply to the deployment.
+
+```
+{
+    // Datacenters is a list of nomad datacenters on which this test deployment
+    // should be scheduled. Nomad supports multiple datacenter deployments. By
+    // default this should be all datacenters.
+    "Datacenters": list of strings,
+
+    // Priority is an integer from [1, 100], the higher the more important. This
+    // allows nomad to determine which tasks should be scheduled when there is
+    // resource contention. If your nomad cluster has other tasks running on it,
+    // be sure to set this value accordingly. Otherwise, a default of 50 will be
+    // provided.
+    "Priority": int,
+}
+```
+
+#### `Deployments: list of objects`
+
```

Comment:
Again, a snippet of the json object for easy parsing, also it allows you to show an example configuration.